src/main/java/com/upserve/uppend/blobs/VirtualAppendOnlyBlobStore.java

```diff
@@ -1,20 +1,34 @@
 package com.upserve.uppend.blobs;
 
+import com.upserve.uppend.metrics.BlobStoreMetrics;
 import org.slf4j.Logger;
 
 import java.lang.invoke.MethodHandles;
 
 public class VirtualAppendOnlyBlobStore extends VirtualPageFileIO {
     private static final Logger log = org.slf4j.LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+    final BlobStoreMetrics.Adders blobStoreMetricsAdders;
+
+
     public VirtualAppendOnlyBlobStore(int virtualFileNumber, VirtualPageFile virtualPageFile) {
+        this(virtualFileNumber, virtualPageFile, new BlobStoreMetrics.Adders());
+    }
+
+    public VirtualAppendOnlyBlobStore(int virtualFileNumber, VirtualPageFile virtualPageFile, BlobStoreMetrics.Adders blobStoreMetricsAdders) {
         super(virtualFileNumber, virtualPageFile);
+        this.blobStoreMetricsAdders = blobStoreMetricsAdders;
     }
 
     public long append(byte[] bytes) {
-        final long pos = appendPosition(recordSize(bytes));
+        final long tic = System.nanoTime();
+        final int size = recordSize(bytes);
+        final long pos = appendPosition(size);
         write(pos, byteRecord(bytes));
         if (log.isTraceEnabled()) log.trace("appended {} bytes to {} at pos {}", bytes.length, virtualFileNumber, pos);
+        blobStoreMetricsAdders.appendCounter.increment();
```

Comment:
How much might using these counters affect performance?

Reply:
Curious to see what total time is for LongAdder.increment in JProfiler...